Functions.Templates/Templates/HttpTrigger-Python/__init__.py

```diff
@@ -1,20 +1,22 @@
 import logging
 
+# Add any dependencies you might need here
 import azure.functions as func
 
 
 def main(req: func.HttpRequest) -> func.HttpResponse:
+    """
+    This is the function that will be called when calling your Azure Function.
+    Currently, it is a template for an HTTP Trigger that says hello when calling your URL.
+    Try changing the code found below to manipulate your function!
+    If you want to change where your Azure Function looks, take a look at 'function.json'
+    """
     logging.info('Python HTTP trigger function processed a request.')
 
+    # Getting the 'name' parameter from the url string
     name = req.params.get('name')
-    if not name:

```

Comment:
There are no quickstarts or tutorials that start with using a POST request with a body.  I don't see why we need this here to further complicate something when users are first starting out.

Reply:
I partially agree - this snippet does help folks to know how to read the body. I think it'll be better to add methods in the function.json (get and post) to fix this issue.

Reply:
So the function.json comes with GET and POST today.  Are you saying that we fix that to just GET? 

Another solution to this could be to have helper functions that we've moved out of this main function dealing with GET and POST requests.  If we want to keep it in, we should be very explicit about what it is and why it's there.  That would be more code but it could help users see what's happening.

Reply:
Compared with the JavaScript HTTP trigger template, https://github.com/Azure/azure-functions-templates/blob/dev/Functions.Templates/Templates/HttpTrigger-JavaScript/index.js#L4

The `name` variable can be assigned from query parameter (GET) or request body (POST).
```javascript
const name = (req.query.name || (req.body && req.body.name));
```
Here, the Python template is just following the same pattern.

Reply:
I don't recommend removing this.